### ISSUE TITLE
Cookies required, but not available in the page pulled up by GCM

### ISSUE THREAD

**Issue by mvonballmo** [GitCredentialManager/git-credential-manager#900]
**Which version of GCM are you using?**

From a terminal, run `git credential-manager-core --version` and paste the output.

2.0.785+6074e4e3d3

**Which Git host provider are you trying to connect to?**

* [x] Azure DevOps
* [ ] Azure DevOps Server (TFS/on-prem)
* [ ] GitHub
* [ ] GitHub Enterprise
* [ ] Bitbucket
* [ ] Other - please describe

**Can you access the remote repository directly in the browser using the remote URL?**

From a terminal, run `git remote -v` to see your remote URL.

* [x] Yes
* [ ] No, I get a permission error
* [ ] No, for a different reason - please describe

---

**_[Azure DevOps only]_ What format is your remote URL?**

* [ ] Not applicable
* [ ] https://dev.azure.com/`{org}`/...
* [x] https://`{org}`@dev.azure.com/`{org}`/...
* [ ] https://`{org}`.visualstudio.com/...

**_[Azure DevOps only]_ If the account picker shows more than one identity as
you authenticate, check that you selected the same one that has access on the
web.**

* [x] Not applicable
* [ ] I only see one identity
* [ ] I checked each identity and none worked

---

**Expected behavior**

I am authenticated and my Git operation completes successfully.

**Actual behavior**

On every remote command, I get the following:

![image](https://user-images.githubusercontent.com/1214927/195509509-e52b8261-8ae9-4a4e-8f1c-d7e816203247.png)

The default browser is Edge.

![image](https://user-images.githubusercontent.com/1214927/195513361-45d2b8cf-a04e-4a8e-b4dd-b19060dc0990.png)

Cookies are set to the default settings there (enabled).

![image](https://user-images.githubusercontent.com/1214927/195513453-c4755844-d0b4-4f59-afd1-4db83784e84a.png)

**Logs**

Set the environment variables `GCM_TRACE=1` and `GIT_TRACE=1` and re-run your
Git command. Review and redact any private information and attach the log.

If you are running inside of Windows Subsystem for Linux (WSL), you must also
set an additional environment variable to enable tracing: `WSLENV=$WSLENV:GCM_TRACE`.
For example:

```shell
WSLENV=$WSLENV:GCM_TRACE:GIT_TRACE GCM_TRACE=1 GIT_TRACE=1 git fetch
```

[git-fetch-log.txt](https://github.com/GitCredentialManager/git-credential-manager/files/9771177/git-fetch-log.txt)

**Comment by ldennington** [comment]
At a glance, looks like [MSAL ](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet) is throwing an error in. In order to unblock you while we investigate further, consider switching to a [different auth mode](https://github.com/GitCredentialManager/git-credential-manager/blob/main/docs/configuration.md#credentialgithubauthmodes).

**Comment by ldennington** [comment]
@mvonballmo Did you try a PAT? I believe that one is available for Azure DevOps but apologies if I'm mistaken.

**Comment by mjcheetham** [comment]
@mvonballmo you can try setting the following to switch to the System/default browser for Azure DevOps auth:

```shell
git config --global credential.msauthFlow system
```

See the documentation here: https://aka.ms/gcm/config#credentialmsauthflow

@ldennington is correct the problem is the embedded webview that MSAL uses is based on Internet Explorer, so there must be some compatibility problems there.

**Comment by mvonballmo** [comment]
@mjcheetham: Thanks for telling me about `credential.msauthFlow`. The login prompt now appears in MS Edge, but ends up on this page after I've provided my username and password.

![image](https://user-images.githubusercontent.com/1214927/196347315-68334b8b-eca4-40a9-812f-b6c90f3b4e59.png)

I wouldn't which server would be running at `http://localhost:56518/`.

For reference, the machine that's giving me trouble is Windows Server 2016. I'm not sure whether there might be some prerequisites for GCM missing on this machine?

**Comment by mvonballmo** [comment]
> @mvonballmo Did you try a PAT? I believe that one is available for Azure DevOps but apologies if I'm mistaken.

I have not tried a PAT, no. I suppose I could, but this is just a workstation for me ... and the authentication was working just fine until quite recently. I'm not sure what's changed, though. After I started getting problems, I upgraded both Git and GCM to the latest released versions.

**Comment by mjcheetham** [comment]
@mvonballmo please could you try running the following command inside of your repository and post back the logs (please redact any sensitive information!):

```sh
git credential-manager-core diagnose
```

> I wouldn't know which server would be running at http://localhost:56518/.

GCM (or more specifically the MSAL library we use internally) will open a port (in this example, port `56518`) and listen for the redirect to capture the `?code=<code>` to resume authentication. A random free port is selected so this number will change each time.

**Comment by mvonballmo** [comment]
> @mvonballmo please could you try running the following command inside of your repository and post back the logs (please redact any sensitive information!):
>
> git credential-manager-core diagnose

[gcm-diagnose.log](https://github.com/GitCredentialManager/git-credential-manager/files/9835998/gcm-diagnose.log)

I hope that helps.

**Comment by mvonballmo** [comment]
Sometime today, I tried fetching from this repository on this machine and it worked, though without asking for authentication. I am unsure what changed. I was able to both `fetch` and `push` and can no longer reproduce this behavior (for now).

**Comment by mjcheetham** [comment]
Hmm... that's weird. The diagnose logs show the loopback (that the system browser flow uses) is working ok:

```text
Diagnostic: Networking
Skipped: False
Success: True
Exception: None
Log:
Checking basic networking and HTTP stack...
Creating HTTP client... OK
IsNetworkAvailable: True
Sending HEAD request to http://example.com... OK
Sending HEAD request to https://example.com... OK
Acquiring free TCP port... OK
Testing local HTTP loopback connections...
Creating new HTTP listener for http://localhost:62463/... OK
Waiting for loopback connection... OK
Writing response... OK
Waiting for response data... OK
Loopback connection data OK
Testing with IHttpClientFactory created HTTP client...
Creating HTTP client... OK
IsNetworkAvailable: True
Sending HEAD request to http://example.com... OK
Sending HEAD request to https://example.com... OK
Acquiring free TCP port... OK
Testing local HTTP loopback connections...
Creating new HTTP listener for http://localhost:62466/... OK
Waiting for loopback connection... OK
Writing response... OK
Waiting for response data... OK
Loopback connection data OK
```

It's possible there was something preventing the loopback connection from working before that's working now.. for example a local firewall, anti-virus, proxy or policy.

Given we can no longer reproduce the problem and it seems to be working for you again, I will close this issue. If the problem reoccurs please open a new issue. Thanks!

**Comment by mvonballmo** [comment]
> The diagnose logs show the loopback (that the system browser flow uses) is working ok

I noticed that as well. I think you're right; something changed that allowed the authentication to complete. Thanks for looking into it.

**Comment by sheian** [comment]
No worries about that.. maybe we should aware for the for the transaction and also careful for the background of our cellphones.. and your always welcome

Sent from my Metro By T-Mobile 4G LTE Android Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by sheian** [comment]
.....................................

Gretchen Lian would like to stay in touch on LinkedIn.

Gretchen Lian Iglesia
Freelance
Philippines

View profile https://www.linkedin.com/comm/in/gretchen-lian-iglesia-742799252?trk=eml-guest-invite-pprofile-link-cta&trkEmail=eml-email_m2g_invite_single_01-null-373-null-null-9ad7b%7Elfv9xgjd%7E9s-null-neptune%2Fprofile%7Evanity%2Eview&lipi=urn%3Ali%3Apage%3Aemail_email_m2g_invite_single_01%3B6pPIrxa%2FQIW0djlqv7G3oA%3D%3D

Confirm that you know Gretchen Lian https://www.linkedin.com/e/v2?e=9ad7b-lfv9xgjd-9s&lipi=urn%3Ali%3Apage%3Aemail_email_m2g_invite_single_01%3B6pPIrxa%2FQIW0djlqv7G3oA%3D%3D&t=suwj&tracking=eml-guest-invite-cta&ek=email_m2g_invite_single_01&validationToken=1AEsmIby&invitationId=7047228834068262912


.....................................

Unsubscribe: https://www.linkedin.com/e/v2?e=9ad7b-lfv9xgjd-9s&t=lun&midToken=AQFmrFVhGEd36Q&ek=email_m2g_invite_single_01&li=381&m=unsub&ts=unsub&loid=AQHU5pEgUPKexAAAAYczJYmzUt058Hv9odgRBkrAPmJZUD3l5DE7ZnIOQcE7giDo2E5LIuaRGe0vspeGSSRTeCovtrL7sue_muCzBBg3U_wRi2GBnIGS19LsnGtwRdyC0rOyQ38uACq46vehgcATIXMSH1CvHhSxmQ&eid=9ad7b-lfv9xgjd-9s

Help: https://www.linkedin.com/e/v2?e=9ad7b-lfv9xgjd-9s&lipi=urn%3Ali%3Apage%3Aemail_email_m2g_invite_single_01%3B6pPIrxa%2FQIW0djlqv7G3oA%3D%3D&a=customerServiceUrl&ek=email_m2g_invite_single_01&li=380&m=footer&ts=help&articleId=67


You are receiving Invitation emails. LinkedIn will use your email address to make suggestions to our members in features like People You May Know.

This email was sent to ***@***.***

© 2023 LinkedIn Ireland Unlimited Company, Wilton Plaza, Wilton Place, Dublin 2. LinkedIn is a registered business name of LinkedIn Ireland Unlimited Company. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.
